src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -424,7 +424,7 @@ public static HaxeClassResolveResult getHaxeClassResolveResult(@Nullable PsiElem
     if (element instanceof HaxeComponentName) {
       return getHaxeClassResolveResult(element.getParent(), specialization);
     }
-    if (element instanceof AbstractHaxeTypeDefImpl) {
+    if (element instanceof HaxeTypedefDeclaration) {
```

Comment:
Why you need this? If I got it right, basically they are the same "typedef", no?